Sources/PostgresNIO/New/Data/Range+PostgresCodable.swift

```diff
@@ -0,0 +1,304 @@
+        postgresRange.encode(into: &byteBuffer, context: context)
+    }
+}
+
+extension Range: PostgresNonThrowingEncodable where Bound: PostgresRangeEncodable {}
+
+extension Range: PostgresDecodable where Bound: PostgresRangeDecodable {
+    @inlinable
+    public init<JSONDecoder: PostgresJSONDecoder>(
+        from buffer: inout ByteBuffer,
+        type: PostgresDataType,
+        format: PostgresFormat,
+        context: PostgresDecodingContext<JSONDecoder>
+    ) throws {
+        let postgresRange = try PostgresRange<Bound>(
+            from: &buffer,
+            type: type,
+            format: format,
+            context: context
+        )
+
+        guard let lowerBound: Bound = postgresRange.lowerBound,
+            let upperBound: Bound = postgresRange.upperBound,
+            postgresRange.isLowerBoundInclusive,
+            !postgresRange.isUpperBoundInclusive
+        else {
+            throw PostgresDecodingError.Code.failure
+        }
+        
+        self = lowerBound..<upperBound
+    }
+}
+
+// MARK: ClosedRange
+
+extension ClosedRange: PostgresEncodable where Bound: PostgresRangeEncodable {
+    public static var psqlType: PostgresDataType { return Bound.psqlRangeType }
+    public static var psqlFormat: PostgresFormat { return .binary }
+
+    @inlinable
+    public func encode<JSONEncoder: PostgresJSONEncoder>(
+        into byteBuffer: inout ByteBuffer,
+        context: PostgresEncodingContext<JSONEncoder>
+    ) {
+        let postgresRange = PostgresRange<Bound>(closedRange: self)
+        postgresRange.encode(into: &byteBuffer, context: context)
+    }
+}
+
+extension ClosedRange: PostgresNonThrowingEncodable where Bound: PostgresRangeEncodable {}
+
+extension ClosedRange: PostgresDecodable where Bound: PostgresRangeDecodable {
+    @inlinable
+    public init<JSONDecoder: PostgresJSONDecoder>(
+        from buffer: inout ByteBuffer,
+        type: PostgresDataType,
+        format: PostgresFormat,
+        context: PostgresDecodingContext<JSONDecoder>
+    ) throws {
+        let postgresRange = try PostgresRange<Bound>(
+            from: &buffer,
+            type: type,
+            format: format,
+            context: context
+        )
+
+        guard let lowerBound: Bound = postgresRange.lowerBound,
+            var upperBound: Bound = postgresRange.upperBound,
+            postgresRange.isLowerBoundInclusive
+        else {
+            throw PostgresDecodingError.Code.failure
+        }
+
+        if !postgresRange.isUpperBoundInclusive {
+            upperBound = try upperBound.upperBoundExclusiveToUpperBoundInclusive()
+        }
+
+        if lowerBound > upperBound {
+            throw PostgresDecodingError.Code.failure
+        }
+        
+        self = lowerBound...upperBound
+    }
+}
+
+// MARK: Private
+@usableFromInline let _isEmpty: UInt8 = 0x01
+@usableFromInline let _isLowerBoundInclusive: UInt8 = 0x02
+@usableFromInline let _isUpperBoundInclusive: UInt8 = 0x04
```

Comment:
Can these go in an enum or struct?

Reply:
See new commit https://github.com/vapor/postgres-nio/pull/330/commits/a6e1dd8666588a946553d4fabd3d41bacb10392d